proxy/freedom/freedom.go

```diff
@@ -383,6 +401,82 @@ func (w *PacketWriter) WriteMultiBuffer(mb buf.MultiBuffer) error {
 	return nil
 }
 
+type NoisePacketWriter struct {
+	Writer     PacketWriter
+	firstWrite bool
+}
+
+// MultiBuffer writer with Noise in first packet
+func (w *NoisePacketWriter) WriteMultiBuffer(mb buf.MultiBuffer) error {
+	if w.firstWrite {
+		w.firstWrite = false
+		//Do not send noise for DNS requests just to be safe
+		if w.Writer.UDPOverride.Port != 53 {
+			var noise []byte
+			var err error
+			//User input string
+			if w.Writer.config.Noise.StrNoise != "" {
+				noise = []byte(w.Writer.config.Noise.StrNoise)
+			} else {
+				//Random noise
+				noise, err = GenerateRandomBytes(randBetween(int64(w.Writer.config.Noise.LengthMin),
+					int64(w.Writer.config.Noise.LengthMax)))
+			}
+
+			if err != nil {
+				return err
+			}
+			_, _ = w.Writer.PacketConnWrapper.Write(noise)
+
+			if w.Writer.config.Noise.DelayMin != 0 {
+				time.Sleep(time.Duration(randBetween(int64(w.Writer.config.Noise.DelayMin), int64(w.Writer.config.Noise.DelayMax))) * time.Millisecond)
```

Comment:
since this is not an actual fragment, is the delay actually making a difference in this case? maybe it can be removed entirely to simplify the PR

Reply:
i'm asking if any difference has been observed in practice.